packages/builder/src/components/integration/QueryViewerSidePanel/index.svelte

```diff
@@ -41,7 +41,7 @@
   </div>
   <div class="content">
     {#if activeTab === "JSON"}
-      <JSONPanel data={rows[0] || {}} />
```

Comment:
This was displaying only the first results on the display, instead of the actual result
<img width="658" alt="image" src="https://github.com/Budibase/budibase/assets/15987277/819dd3d6-4465-4681-90d7-43e211a9e592">

Reply:
This was by design originally - it was supposed to show what a "row" would look like returned from the query. Printing the whole response could be really slow if you have 50K+ rows as theres no virtual rendering.

Reply:
I did raise an issue nearly two years ago relating to this: https://github.com/Budibase/budibase/issues/5591

Closed it because it was stale, but still a thing I suppose.

Reply:
We should then trim the response, instead of showing the first row. Otherwise, it will display that the response is an object instead of an array

Reply:
I left all the reviews as they were before, except when the result returns multiple rows. In this case, we will show that is an array, but showing a max of 5 results followed by `...`
<img width="1054" alt="image" src="https://github.com/Budibase/budibase/assets/15987277/d84b2d53-2e49-4bb7-a99d-0d7c8e7d34cb">

Reply:
Done @melohagan 
<img width="471" alt="image" src="https://github.com/Budibase/budibase/assets/15987277/0e2f1891-27a8-4f24-aa29-a3e7259396d3">

Reply:
![Screenshot 2024-02-08 at 13 46 10](https://github.com/Budibase/budibase/assets/101575380/6c2ec1e7-f97b-417f-9c1d-4825a101d9cd)

Also noticed that the `... x further items` does not work recursively, so nested arrays that have more than five objects show all.

Reply:
> ![Screenshot 2024-02-08 at 13 46 10](https://private-user-images.githubusercontent.com/101575380/303352242-6c2ec1e7-f97b-417f-9c1d-4825a101d9cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0MDE0MjIsIm5iZiI6MTcwNzQwMTEyMiwicGF0aCI6Ii8xMDE1NzUzODAvMzAzMzUyMjQyLTZjMmVjMWU3LWY5N2ItNDE3Zi05YzFkLTQ4MjVhMTAxZDljZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwOFQxNDA1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYWZlMmU5YjM0ZTIxMmNiYTM3NmZjNjI0NDI5NWEwOWM0YTQ0YTA4YTY1N2NmZjM3ZTk0ZjMyYjg2NzY1YzRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5t-xpb4_2EI6ZrLeo46cH35wLMR_RSf1Xmw2baWuN9k)
> 
> Also noticed that the `... x further items` does not work recursively, so nested arrays that have more than five objects show all.

I think recursively working this would be too complex - realistically if a user has an array within a row with hundreds of elements, we are going to have to display it no matter what, whether it was the old way or the new way of displaying this.

Reply:
Fixed! 
![Uploading image.png…]()

Reply:
Looks great! 🥇